### PR TITLE
The banishment of Address

### PR DESCRIPTION
**Pull request by cloutiertyler**
# Description of Changes

Jeff and I worked on Friday to remove all uses of `Address` as an identifier for a database. `Address` is still used for what will eventually become `ConnectionId`.

There is a corresponding [private PR](https://github.com/clockworklabs/SpacetimeDBPrivate/pull/1067).

# API and ABI breaking changes

This breaks the CLI API as well as the HTTP and WebSocket APIs. It also edits the system tables.

# Expected complexity level and risk

4 

*This complexity rating applies not only to the complexity apparent in the diff,
but also to its interactions with existing and future code.*

This impacts many APIs as mentioned above. The change is mostly mechanical, but not entirely.

# Testing

*Describe any testing you've done, and any testing you'd like your reviewers to do,
so that you're confident that all the changes work as expected!*

- [x] We have run the unit tests and smoke tests.

### CODE REVIEW

File: crates/core/src/db/datastore/locking_tx_datastore/datastore.rs
Comment (kim) [inline crates/core/src/db/datastore/locking_tx_datastore/datastore.rs:73]: Missed one s/database_address/database_identity

File: crates/core/src/db/datastore/locking_tx_datastore/datastore.rs
Comment (kim) [inline crates/core/src/db/datastore/locking_tx_datastore/datastore.rs:101]: Missed one

File: crates/core/src/db/relational_db.rs
Comment (kim) [inline crates/core/src/db/relational_db.rs:328]:
```suggestion
                    "mismatched database identity: {} != {}",
```

File: crates/core/src/db/relational_db.rs
Comment (kim) [inline crates/core/src/db/relational_db.rs:497]:
```suggestion
    /// Returns the identity of this database
```

File: crates/core/src/host/module_host.rs
Comment (kim) [inline crates/core/src/host/module_host.rs:185]:
```suggestion
    /// The identity of the database.
```

File: tools/clippy.sh
Comment (kim) [inline tools/clippy.sh:9]: Uncomment before merge?

File: crates/core/src/auth/identity.rs
Comment (kim) [inline crates/core/src/auth/identity.rs:23]: Is this related to this patch?

**Review by kim — APPROVED** [review]
lgtm

File: crates/client-api/Cargo.toml
Comment (jsdt) [reply crates/client-api/Cargo.toml:45]: It might be worth switching soon, but I don't think I can do that before the next release.

File: crates/core/src/auth/identity.rs
Comment (jsdt) [reply crates/core/src/auth/identity.rs:23]: The diff appears to be showing changes that were merged in https://github.com/clockworklabs/SpacetimeDB/pull/1854.

**Review by gefjon — APPROVED** [review]
Trusting kim, just in case I somehow wound up as a code owner on this. I have not actually reviewed this PR, but Kim's review satisfies me.

File: crates/cli/src/subcommands/call.rs
Comment (jdetter) [inline crates/cli/src/subcommands/call.rs:23]: Note for @bfops we should probably do some code deduplication on this param because it's used quite frequently it seems.

**Review by jdetter — APPROVED** [review]
I have not tested this but I've reviewed the files I'm a code-owner of. I left one follow-up note for Zeke but these changes look correct to me.

**Comment by cloutiertyler** [comment]
Although I am a partial author of this, I just want to so that this one looks good to me.

**Pull request merged**
